### PR TITLE
build(all): migrate to eslint

### LINKED ISSUE #1241: [CHORE] Replace tslint with eslint

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
- Imports are now automatically grouped and sorted alphabetically. 2 or more destructured imports will automatically be multiline.
- Trailing commas are now required for multiline arrays, objects, and parameter lists. This will be automatically fixed by the linter.
- Typecasts now must use `<>`.
- Indents are spaces, size of 2. I'm fine with changing this but `.editorconfig` must be updated accordingly. 2 space is the current setting.
- Arrow functions that are just a single return must be lambda style. This is auto fixed.
- I disabled member ordering for now but it would be nice to choose an order that we can adhere to.
- Unused imports are warnings since this is not auto fixable.

Fixes #1241

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            ]
+          }
+        ],
+        '@typescript-eslint/consistent-type-assertions': [
+          'error',
+          {
+            assertionStyle: 'angle-bracket'
+          }
+        ],
+        '@typescript-eslint/consistent-type-definitions': 'error',
```
Comment (lderrickable) [inline .eslintrc.js:107]: I don't quite understand this one. Isn't it supposed to be either `[ 'error', 'interface']` or `['error', 'type']`?

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-definitions.md

Comment (griest024) [reply .eslintrc.js:107]: I think the `interface` option is the default, but I can set it explicitly if you want

Comment (griest024) [reply .eslintrc.js:107]: for example 
```ts
type Foo = {
  thing: string
}
```
will trigger a linting error with the current setting

Comment (griest024) [reply .eslintrc.js]: changed

File: libs/category/src/drivers/injection-tokens/category-driver.token.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { InjectionToken } from '@angular/core';
 
+import { DaffProduct } from '@daffodil/product';
+
 import { DaffCategoryServiceInterface } from '../interfaces/category-service.interface';
 
 //TODO(damienwebdev): This import is necessary until we ship Ivy packages, do not remove it.
 //See: https://github.com/ng-packagr/ng-packagr/issues/1844
```
Comment (lderrickable) [inline libs/category/src/drivers/injection-tokens/category-driver.token.ts:8]: this todo needs to move too

**Comment by lderrickable** [comment]
lgtm after that last change request. Though next time I think we should do the auto-lint and the manual linting changes in separate PRs.

File: libs/category/src/drivers/injection-tokens/category-driver.token.ts (hunk 2 above)
Comment (griest024) [reply libs/category/src/drivers/injection-tokens/category-driver.token.ts]: I think this is going to be removed very soon anyway but changed

**Comment by griest024** [comment]
@lderrickable thanks for the review

**Comment by lderrickable** [comment]
@griest024 and thanks for the migration

**Review by lderrickable — DISMISSED** [review]

**Comment by lderrickable** [comment]
Needs approved by @damienwebdev

**Comment by damienwebdev** [comment]
Commits need a suffix for rebase merging and appropriate messages for the last few commits, otherwise LGTM.

**Comment by griest024** [comment]
@damienwebdev changed

**Pull request merged**